docker/developers/docker-compose.commands.yml

```diff
@@ -0,0 +1,211 @@
+    extends:
+      file: docker-compose.dashboard.yml
+      service: web
+    volumes: &common-shell-volumes
+      - home:/mnt/home
+    command: /bin/bash -l
+  shell-aws: &command-aws-base
+    extends:
+      file: docker-compose.dashboard.yml
+      service: web-aws
+    volumes: *common-shell-volumes
+    command: /bin/bash -l
+
+  # Spins up an environment to authenticate a Code.org staff engineer's
+  # AWS credentials.
+  aws-access:
+    <<: *command-aws-base
+    command: /bin/bash -l -c "./bin/aws_access"
+
+  # Spins up a dashboard server console with contributor access
+  console:
+    <<: *command-base
+    environment:
+      SKIP_I18N_INIT: true
+    command: /bin/bash -l -c './bin/dashboard-console'
+  console-full-init:
+    <<: *command-base
+    command: /bin/bash -l -c './bin/dashboard-console'
+  console-aws:
+    <<: *command-aws-base
+    environment:
+      SKIP_I18N_INIT: true
+    command: /bin/bash -l -c './bin/dashboard-console'
+  console-aws-full-init:
+    <<: *command-aws-base
+    command: /bin/bash -l -c './bin/dashboard-console'
+
+  # Open the MySQL client on the web instance to connect to the database
+  # service running in the other container.
+  mysql-client-admin:
+    <<: *command-base
+    command: /bin/bash -l -c './bin/mysql-client-admin'
+
+  # Run the 'rake build' step with/without AWS
+  build:
+    <<: *command-base
+    command: /bin/bash -i -c "bundle exec rake build"
+  build-aws:
+    <<: *command-aws-base
+    command: /bin/bash -i -c "bundle exec rake build"
+
+  # Run any rake task
+  rake:
+    <<: *command-base
+    entrypoint: ['/bin/bash', '-l', './docker/developers/run.sh', 'rake']
+    command: ['--tasks']
+
+  # Run any rake task within the dashboard project
+  dashboard-rake:
+    <<: *command-base
+    working_dir: /app/src/dashboard
+    entrypoint: ['/bin/bash', '../docker/developers/run.sh', 'rake']
+    command: ['--tasks']
+
+  # Run any rake task within the pegasus project
+  pegasus-rake:
+    <<: *command-base
+    working_dir: /app/src/pegasus
+    entrypoint: ['/bin/bash', '../docker/developers/run.sh', 'rake']
+    command: ['--tasks']
+
+  # Run any database migrations
+  migrations:
+    <<: *command-base
+    command: /bin/bash -l -c 'cd dashboard && ../docker/developers/run.sh bundle exec rake db:migrate && RAILS_ENV=test ../docker/developers/run.sh bundle exec rake db:migrate && cd ../pegasus && ../docker/developers/run.sh bundle exec rake db:migrate'
+
+  lint-ruby:
+    <<: *command-base
+    entrypoint: ['/bin/bash', './docker/developers/run.sh', 'bundle', 'exec', 'rubocop']
+    command: []
+
+  # Builds the JavaScript for the site
+  build-js:
+    <<: *command-isolated-base
+    volumes:
+      - nvm:/home/cdodev/.nvm
+      - ../..:/app/src
+    working_dir: /app/src/apps
+    command: /bin/bash -l -c '../docker/developers/run.sh yarn build'
+
+  # Runs a server that watches for changes made to JS code and rebuilds on update
+  build-js-server:
```

Comment:
Should this be `run-js-server`?

Reply:
But it builds? It is the `build-js` step but as a sustained server. It's a little confusing to have it be `run`, no?